core/build.gradle

```diff
@@ -62,9 +62,12 @@ tasks.withType(Jar) {
 // Proguard can't handle multi-release jars so we recreate it here.
 tasks.trimShadedJar.doLast {
     // outjars is a file, so only one jar generated for sure
-    def trimmed = tasks.trimShadedJar.outJarFiles[0].toPath()
+    def outJarFile = tasks.trimShadedJar.outJarFiles[0]
+    def trimmed = outJarFile.toPath().toString()
+    def tmp = "${trimmed}.tmp"
 
-    ant.jar(destfile: trimmed.toString(), update: true, duplicate: 'fail') {
+    ant.jar(destfile: tmp, duplicate: 'fail', filesetmanifest:'merge') {
+        zipfileset(src: outJarFile, excludes: 'META-INF/services/java.security.Provider')
```

Comment:
How about adding a new property `shadowExclusions` like `jacocoExclusions` and excluding `shadowExclusions` in `shadeJar`?
https://github.com/line/armeria/blob/6dec8498457d7032cbbc192e4070ca2bc5658535/gradle.properties#L29

Reply:
In order to do so, I think:
- This functionality should be applied to all modules, not just `:core`. If we take this approach, it probably makes more sense to move this functionality to `gradle-scripts`
- In the case of `:core`, the filtering will need to happen twice: once to add `versions`, another time to remove `shadowExclusions`

Am I understanding your intention correctly?

Reply:
I didn't think we needed `shadowExclusions` per module. I imagined adding `shadowExclusions` in `java-shade.gradle`. For example:
```groovy
task shadedJar(

	def shadowExclusions = []
	if (rootProject.hasProperty('shadowExclusions')) {
	    shadowExclusions = rootProject.findProperty('shadowExclusions').split(",")
	}
	shadowExclusions.each {
	    exclude it
	}
}
```

Reply:
My proposal intended to use the feature provided by Shadow instead of the workaround.
If you think you think a different exclusion is required for each module, [`bomGroups`](https://github.com/line/gradle-scripts#generating-maven-bom-with-bom-flag) pattern is also good to me.

Reply:
I see, this solution also works 👍  Note that I wasn't able to migrate the other exclusions to ensure backwards compat

Reply:
Sounds good. Let's regard them as the default exclusions.